paimon-core/src/main/java/org/apache/paimon/casting/CastExecutors.java

```diff
@@ -215,19 +245,52 @@ public class CastExecutors {
                                             (((Timestamp) value).getMillisecond()
                                                     % DateTimeUtils.MILLIS_PER_DAY);
                         }
+                    case TINYINT:
+                        return value -> (byte) (((Timestamp) value).getMillisecond());
```

Comment:
Is `getMillisecond` correct? You can take a look to Flink and Spark.

Reply:
done.